spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -153,7 +153,14 @@ public void nack() {
 		}
 		catch (RuntimeException re) {
 			if (this.ackMode == AckMode.AUTO) {
-				message.nack();
+				if (getErrorChannel() != null) {
```

Comment:
Unit testing this logic would be great, but can be done in a follow up, so we can have the error handling in (today's?) release.

Reply:
Sounds good, will implement shortly after clarification with Artem on whether we can actually revert this or not.